src/gar.lua

```diff
@@ -1848,8 +1848,8 @@ function gar.redelegateStake(params)
 	GatewayRegistry[targetAddress] = targetGateway
 
 	return {
-		sourceGateway = sourceGateway,
-		targetGateway = targetGateway,
+		sourceGateway = utils.deepCopy(sourceGateway, { "delegates", "vaults", "settings.allowedDelegatesLookup" }),
+		targetGateway = utils.deepCopy(targetGateway, { "delegates", "vaults", "settings.allowedDelegatesLookup" }),
 		redelegationFee = redelegationFee,
 		feeResetTimestamp = currentTimestamp + constants.redelegationFeeResetIntervalMs,
 		redelegationsSinceFeeReset = Redelegations[delegateAddress].redelegations,
```

Comment:
nit: can share state above. `local redelegationsSinceFeeReset = (previousRedelegations and previousRedelegations.redelegations or 0) + 1` onto ~L1838